api/v1beta2/awsmachine_types.go

```diff
@@ -255,6 +260,40 @@ type AWSMachineSpec struct {
 	CapacityReservationPreference CapacityReservationPreference `json:"capacityReservationPreference,omitempty"`
 }
 
+// DynamicHostAllocationSpec defines the configuration for dynamic dedicated host allocation.
+type DynamicHostAllocationSpec struct {
+	// InstanceFamily specifies the EC2 instance family (e.g., "m5", "c5", "r5").
+	// +kubebuilder:validation:Required
+	InstanceFamily string `json:"instanceFamily"`
```

Comment:
can we have regex cel/webhook validation to prevent invalid input for InstanceFamily / InstanceType.

Can we also compute the upper level spec.InstanceType in that validation, so any invalid input fails earlier

Reply:
the host family is now inferred from the machine spec